app/src/main/kotlin/at/bitfire/davdroid/ui/AppSettingsScreen.kt

```diff
@@ -492,15 +506,118 @@ fun AppSettings_Integration(
     )
 
     val context = LocalContext.current
+    val pm = context.applicationContext.packageManager
+
+    var distributor by remember { mutableStateOf(UnifiedPush.getSavedDistributor(context)) }
+    val distributors = remember { UnifiedPush.getDistributors(context) }
```

Comment:
I think we should take the data from the model:

- For instance if the methods block, they would block the UI.
- Also the UI shouldn't be coupled to business logic for modularity.